### PR TITLE
tmt: Add TMT_TEST env variable to specify single test to run

### PR DESCRIPTION
**Pull request by ckyrouac**
e.g. `TMT_TEST=test-21-logically-bound-switch make test-tmt` will run all the tmt setup code then run test-21-logically-bound-switch. This is useful when developing a test.

### CODE REVIEW

**Comment by cgwalters** [comment]
Yes! I was literally looking for this 10 minutes ago :smile:

File: xtask/src/xtask.rs [hunk 1]
```diff
@@ -186,14 +186,26 @@ fn test_tmt(sh: &Shell) -> Result<()> {
...
+}
+
+#[context("test-integration")]
+fn test_tmt(sh: &Shell) -> Result<()> {
+    let tests = match std::env::var("TMT_TEST") {
+        Ok(test) => {
+            vec![(0, test)]
+        }
+        Err(_) => all_plan_files(sh)?,
+    };
```
Comment (cgwalters) [inline xtask/src/xtask.rs:199]: I think this can be `let tests = std::env::var("TMT_TEST").unwrap_or_else(|| vec![(0, test]));`

**Review by cgwalters — APPROVED** [review]
OK as is, just an optional suggestion

File: xtask/src/xtask.rs [hunk 2]
```diff
@@ -186,14 +186,26 @@ fn test_tmt(sh: &Shell) -> Result<()> {
...
+
+    println!("Discovered plans: {tests:?}");
```
Comment (cgwalters) [inline xtask/src/xtask.rs:201]: Now when `TMT_TEST` is set we'll now only say that we discovered what they asked for right?

How about doing this like:
```
let mut tests = all_plan_files()?;
if let Ok(name) std::env::var("TMT_TEST") {
    tests.retain(|x| x.1.as_str() == name);
    if tests.is_empty() {
        anyhow::bail!("Failed to match test: {name}");
    }
}
```

or so

Comment (ckyrouac) [reply xtask/src/xtask.rs]: good idea, updated

**Pull request merged**